delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/TimeRangeSuite.scala

```diff
@@ -0,0 +1,70 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.search
+
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.TimeRange.ParseError.{InvalidFormat, InvalidRange, InvalidValue}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.TimeRange.{parse, After, Anytime, Before, Between}
+import ch.epfl.bluebrain.nexus.testkit.bio.EitherAssertions
+import munit.FunSuite
+
+import java.time.Instant
+
+class TimeRangeSuite extends FunSuite with EitherAssertions {
+
+  private val december_2020_string = "2020-12-20T20:20:00Z"
+  private val december_2020        = Instant.parse(december_2020_string)
+  private val april_2022_string    = "2022-04-10T04:20:00Z"
+  private val april_2022           = Instant.parse(april_2022_string)
+
+  List(
+    "",
+    "FAIL",
+    ".."
+  ).foreach { badInput =>
+    test(s"Fail to parse '$badInput' for an invalid format") {
+      parse(badInput).assertLeftOf[InvalidFormat]
+    }
+  }
+
+  List(
+    "BAD..*",
+    s"BAD..$december_2020_string",
+    "*..BAD",
+    s"$december_2020_string..BAD",
+    "2020..*",
+    "2020-12-20..*",       // Missing the time
+    "2020-12-20T20:20..*", // Missing the second and the timezone
+    "2020-12-20T20:20Z..*" // Missing the timezone
+  ).foreach { badInput =>
+    test(s"Fail to parse '$badInput' for an invalid value") {
+      parse(badInput).assertLeftOf[InvalidValue]
+    }
+  }
+
+  test("Parse '*..*' as anytime") {
+    parse("*..*").assertRight(Anytime)
+  }
+
+  test("Parse None as anytime") {
+    parse(None).assertRight(Anytime)
+  }
```

Comment:
I wonder whether the option version of this method is needed. It makes test coverage a bit weird because there are no Some() conditions currently tested

Perhaps a better approach is to just have parse(string) and do the option unpacking elsewhere. This is probably a bit more consistent with a standard parse operation also